src/OSPSuite.Core/Domain/Builder/NeighborhoodBuilder.cs

```diff
@@ -19,15 +19,32 @@ public interface INeighborhoodBase : IContainer
 
    public class NeighborhoodBuilder : Container, INeighborhoodBase
    {
-      //We define a property set for the first neighbor only to be compatible with serialization prior to v12
+      private ObjectPath _firstNeighborPath;
+      private ObjectPath _secondNeighborPath;
+
       public IContainer FirstNeighbor { get; private set; }
 
-      //We define a property set for the second neighbor only to be compatible with serialization prior to v12
       public IContainer SecondNeighbor { get; private set; }
 
-      public ObjectPath FirstNeighborPath { get; set; }
+      public ObjectPath FirstNeighborPath
+      {
+         get => _firstNeighborPath;
+         set
+         {
+            _firstNeighborPath = value;
+            FirstNeighbor = null;
```

Comment:
Should we reset the neighbor only if the new value for the FirstNeighborPath is different from the old value?